internal/server/httpserver/middlewares/hashmiddleware/hashmiddleware.go

```diff
@@ -0,0 +1,120 @@
+// Package hashmiddleware реализует обертку над http.Request и http.ResponseWriter, которая считает и проверяет валидность
+// хеша в запросе и выдает в ответе. Хеш передается в заголовке HashSHA256
+//
+// Package hashmiddleware implements a wrapper around http.Request and http.ResponseWriter that calculates and checks the validity
+// hash in the request and issues a response. The hash is passed in the HashSHA256 header
+package hashmiddleware
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"net/http"
+
+	"go.uber.org/zap"
+
+	"github.com/h2p2f/practicum-metrics/internal/server/servererrors"
+)
+
+// CheckDataHash  - функция для проверки хеша данных запроса
+//
+// CheckDataHash - function to check hash of request data
+func checkDataHash(checkSum string, key string, data []byte) (bool, error) {
```

Comment:
ключ не используется, зачем он?

Reply:
```suggestion
func checkDataHash(checkSum string, key string, data []byte) (bool, error) {
```
```suggestion
if checkSum != "" && key != "" {
```
Я тащу сюда ключ только для проверки необходимости вычисления хеша

Reply:
сорри за запутанный коммент, еще не до конца разобрался в принципах оформления

Reply:
Это если комментатор хочет предложить изменения которые можно внести просто по кнопке согласиться